src/app/vault/ciphers.component.html

```diff
@@ -30,15 +28,15 @@
                 </div>
             </a>
         </div>
-        <div class="no-items" *ngIf="!filteredCiphers.length">
-            <i class="fa fa-spinner fa-spin fa-3x" *ngIf="!loaded" aria-hidden="true"></i>
-            <ng-container *ngIf="loaded">
-                <i class="fa fa-frown-o fa-4x" aria-hidden="true"></i>
-                <p>{{'noItemsInList' | i18n}}</p>
-                <button (click)="addCipher()" class="btn block primary link">{{'addItem' | i18n}}</button>
-            </ng-container>
-        </div>
-    </ng-container>
+    </cdk-virtual-scroll-viewport>
+    <div class="no-items" *ngIf="!ciphers.length">
```

Comment:
The `cdk-virtual-scroll-viewport` is only a container for managing the virtually scrolling list, so there's no reason for this `<div>` to be inside it.

Regarding the old `ng-container` more generally, it did 2 things:
1. `*ngIf="ciphers"` - but that seems unnecessary because this component always instantiates an empty array in the declaration of the `ciphers` property
2. assigned the value of `ciphers`/`pagedCiphers` to `filteredCiphers` - but that is not needed any more because there are no `pagedCiphers`.

That's why I deleted it.

Reply:
thanks, makes sense